modules/_labs/menu/react/lib/Menu.tsx

```diff
@@ -39,7 +39,7 @@ export interface MenuProps extends GrowthBehavior, React.HTMLAttributes<HTMLULis
   /**
    * The HTML `id` of the element that labels the Menu. Often used with menu buttons.
    */
-  labeledBy?: string;
+  menuArialabeledBy?: string;
```

Comment:
Do we need the component name included here? We could also consider falling back to the default `aria-labeledby` prop here as well. 

#526 only covers one use case, yet you've altered several other cases here. I feel like there needs to be more discussion on the direction and overall strategy here before we get to reviewing this much further. Could you please formulate and document your own opinion/reasoning on #526 for each use case before we start that discussion so we have a jumping off point?